anybox/recipe/odoo/runtime/session.py

```diff
@@ -330,22 +350,28 @@ def update_modules_list(self):
 
         This is necessary prior to install of any new module.
         """
-        self.registry('ir.module.module').update_list(self.cr, self.uid)
+        if hasattr(odoo, 'api'):
+            self.env['ir.module.module'].update_list()
+        else:
+            self.registry('ir.module.module').update_list(self.cr, self.uid)
 
     def init_cursor(self):
         db = getattr(self._registry, 'db', None)
         if db is None:  # current trunk (future v8)
             self.cr = self._registry.cursor()
         else:
-            # In OpenERP < 8, Registry.cursor() object is
+            # In Odoo < 8, Registry.cursor() object is
             # a context manager providing auto closing,
             # but we don't want to control the whole lifespan
             # of the cursor.
             self.cr = db.cursor()
 
     def registry(self, model):
         """Lookup model by name and return a ready-to-work instance."""
-        return self._registry.get(model)
+        if hasattr(odoo, 'api'):
+            return self.env[model]
+        else:
+            return self._registry.get(model)
```

Comment:
@StefanRijnhart I'm not sure what motivate that change?

I've a case on an odoo 8.0 project where our developer use in ``upgrade.py`` script the old api while developing, in that case he expected that:
*  ``session.registry('model')`` return the old style model
* ``session.env['model']`` return the new style one

in version 8.0 ``hasattr(odoo, 'api')`` return ``True`` but ``hasattr(odoo, 'env')`` return ``False`` => line 372 failed.

I guess it's something about if this method is called on odoo 10 project?

I have to launch integration test on odoo8/9/10 and add unit test if needs, I was testing directly on a customer project my current fix looks like

```
         """Lookup model by name and return a ready-to-work instance."""
-        if hasattr(odoo, 'api'):
+        if hasattr(odoo, 'env'):
             return self.env[model]
```

Let me know if you have more context about that one!